rest/server_context.go

```diff
@@ -1147,6 +1148,70 @@ func (sc *ServerContext) updateCalculatedStats() {
 
 }
 
+func initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath string) (*gocbcore.Agent, error) {
+	authenticator, err := base.GoCBCoreAuthConfig(clusterUser, clusterPass, certPath, keyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsRootCAProvider, err := base.GoCBCoreTLSRootCAProvider(caCertPath)
```

Comment:
Are we planning to add an unsupported config for testing with self-signed certs?

Reply:
Ultimately yes but as I understand that was going to be part of CBG-1518?